### PR TITLE
Propagation of close event in pair of WebsocketConnection/Connection

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka unfortunately integration tests are for browser only and a test for this bug can be only integration cause it includes the communication between Connection and WebsocketConnection.
Please review.

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@zazabe please review

File: src/websocket-connection.js [hunk 1]
```diff
@@ -116,14 +116,20 @@ WebsocketConnection.prototype.close = function() {
...
     if (this._webSocket.terminate) {
       this._webSocket.terminate();
     } else {
       this._webSocket.close();
     }
-    this.emit('close');
-    resolve();
-  }.bind(this));
+  }
+  this.emit('close');
```
Comment (zazabe) [inline src/websocket-connection.js]: should it be emitted only if the connection is really closed?

Comment (vogdb) [inline src/websocket-connection.js]: The connection can be already closed when calling this method. This method is used for the case when `.close()` is called by client and for the case when `._close()` is called by closed `_webSocket`.

Comment (zazabe) [inline src/websocket-connection.js]: ok, but i meant that we should probably not emit the "close" event if the connection is already closed.

Comment (vogdb) [inline src/websocket-connection.js]: we should for the case when `_webSocket` was the initiator of close.

Comment (zazabe) [inline src/websocket-connection.js]: a ok, i see now that it will be triggered only once if `.close()` is called multiple time, sgtm

**Comment by zazabe** [comment]
lgtm

**Pull request merged**